x-pack/plugins/ai_infra/product_doc_base/server/routes/installation.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter } from '@kbn/core/server';
+import {
+  INSTALLATION_STATUS_API_PATH,
+  INSTALL_ALL_API_PATH,
+  UNINSTALL_ALL_API_PATH,
+  InstallationStatusResponse,
+  PerformInstallResponse,
+  UninstallResponse,
+} from '../../common/http_api/installation';
+import type { InternalServices } from '../types';
+
+export const registerInstallationRoutes = ({
+  router,
+  getServices,
+}: {
+  router: IRouter;
+  getServices: () => InternalServices;
+}) => {
+  router.get(
+    {
+      path: INSTALLATION_STATUS_API_PATH,
+      validate: false,
+      options: {
+        access: 'internal',
+        security: {
+          authz: {
+            requiredPrivileges: ['manage_llm_product_doc'],
+          },
+        },
+      },
+    },
+    async (ctx, req, res) => {
+      const { installClient, documentationManager } = getServices();
+      const installStatus = await installClient.getInstallationStatus();
+      const { status: overallStatus } = await documentationManager.getStatus();
+
+      return res.ok<InstallationStatusResponse>({
+        body: {
+          perProducts: installStatus,
+          overall: overallStatus,
+        },
+      });
+    }
+  );
+
+  router.post(
+    {
+      path: INSTALL_ALL_API_PATH,
+      validate: false,
+      options: {
+        access: 'internal',
+        security: {
+          authz: {
+            requiredPrivileges: ['manage_llm_product_doc'],
+          },
+        },
+        timeout: { idleSocket: 20 * 60 * 1000 }, // install can take time.
+      },
+    },
+    async (ctx, req, res) => {
+      const { documentationManager } = getServices();
+
+      await documentationManager.install({
+        request: req,
+        force: false,
+        wait: true,
+      });
+
+      return res.ok<PerformInstallResponse>({
```

Comment:
This will incorrectly report `installed: true` if the task fails.
Steps:
1. Change the `artifactRepositoryUrl` to something that will not resolve
2. Navigate to o11y ai assistant settings
3. Click the `Install` button to invoke this endpoint
4. Wait for the request to complete. It will eventually return a successful response after TM exhausts all retry attempts.

Reply:
Yeah.. working with remote task execution is quite awkward. The issue I have is that once executed, the task instance is deleted, so we can't even retrieve the output status from it. But maybe in case of error the instance is preserved... idk

Reply:
So, I fixed it in https://github.com/elastic/kibana/pull/194379/commits/b9145effcdc675f3e97fc25339e1332f5b3a4c97. 

What I learned though, is that in case of task failure, the failure isn't stored anywhere (the task instance object still gets deleted at the end of the run), so 
1. I have to manually detect if the task was successful or not (by checking install status after the run)...
2. meaning I can't surface the failure cause to the UI...

Reply:
@pgayvallet nice, I think that'll work well enough given our constraints. Thanks for digging into this.